File: backends/dotnet/src/Zappy.Adapters.Persistence/Seed/SeedFiles.cs
namespace Zappy.Adapters.Persistence;

public sealed record SeedMoney(int Amount, string Currency);

public sealed record SeedCategory(string Id, string Name, string Slug);

public sealed record SeedProduct(
    string Id,
    string Name,
    string Slug,
    string Description,
    SeedMoney Price,
    string CategorySlug,
    int Stock,
    string? ImageUrl);

public sealed record SeedPromotionCode(
    string Code,
    string Kind,
    int? Percentage,
    SeedMoney? Amount,
    SeedMoney? MinimumSubtotal,
    DateTimeOffset ValidFrom,
    DateTimeOffset ValidUntil,
    int? UsageLimit,
    int TimesUsed);

public sealed record SeedCustomer(
    string Id,
    string Email,
    string Name,
    string Password,
    DateTimeOffset CreatedAt,
    IReadOnlyList<string> Wishlist);

File: backends/kotlin/zappy-adapters/src/main/kotlin/nl/zappymart/adapters/seed/SeedFiles.kt
package nl.zappymart.adapters.seed

import java.time.Instant

data class SeedMoney(val amount: Int, val currency: String)

data class SeedCategory(val id: String, val name: String, val slug: String)

data class SeedProduct(
    val id: String,
    val name: String,
    val slug: String,
    val description: String,
    val price: SeedMoney,
    val categorySlug: String,
    val stock: Int,
    val imageUrl: String?,
)

data class SeedPromotionCode(
    val code: String,
    val kind: String,
    val percentage: Int?,
    val amount: SeedMoney?,
    val minimumSubtotal: SeedMoney?,
    val validFrom: Instant,
    val validUntil: Instant,
    val usageLimit: Int?,
    val timesUsed: Int,
)

data class SeedCustomer(
    val id: String,
    val email: String,
    val name: String,
    val password: String,
    val createdAt: Instant,
    val wishlist: List<String>,
)
